### PR TITLE
Class

### PR DESCRIPTION
**Pull request by tommyjcarpenter**
No API change. Just reorganizes things to use a class (ABC)

### CODE REVIEW

**Review by jperozo — COMMENTED** [review]
See comments

File: merqube_client_lib/templates/equity_baskets/multieb_create.py [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+        return constituents
+
+    def _inline_to_tp(self, portfolio: dict[str, Any]) -> TargetPortfoliosDates:
+        """
+        Convert the inline portfolio to a list of target portfolios
+        """
+
+        target_portfolios: TargetPortfoliosDates = []
+
+        uom = PortfolioUom.SHARES if portfolio["quantity_type"] == "SHARES" else PortfolioUom.WEIGHT
```
[the remark this answers is not in the document]
Comment (tommyjcarpenter) [reply merqube_client_lib/templates/equity_baskets/multieb_create.py:68]: yeah this is enforced API side. but no enums over the wire

File: merqube_client_lib/templates/equity_baskets/decrement_create.py [hunk 2]
```diff
@@ -61,30 +61,33 @@ def _validate_dec_params(client: MerqubeAPIClient, index_info: ClientDecrementCo
...
+    """create a decrement index on top of a SSTR"""
 
-    index_info = cast(ClientDecrementConfig, index_info)
+    def create(self, config: dict[str, Any], prod_run: bool = False, poll: int = 0) -> CreateReturn:
+        """
+        Creates a new Equity Basket with multiple entries
+        This class does not handle Corax.
+        """
+        client, template, index_info, inner_spec = self._load_template(
+            template_name="DECREMENT_TEMPLATE_VERSION_1", config=config, model=ClientDecrementConfig
```
[the remark this answers is not in the document]
Comment (tommyjcarpenter) [reply merqube_client_lib/templates/equity_baskets/decrement_create.py:73]: well the current configs are overrides for these templates, and if the users want, they can just create a whole manifest. the current templates are like "currated"

File: merqube_client_lib/templates/equity_baskets/multieb_create.py [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+            dates.add(pd.Timestamp(constituent["date"]))
+
+        for date in sorted(dates):
+            positions: list[BasketPosition | RicEquityPosition] = []
+            for constituent in const:
+                sec_type = constituent["security_type"]
+                pos_class = RicEquityPosition if sec_type == "EQUITY" else BasketPosition
+
+                if pd.Timestamp(constituent["date"]) != date:
+                    continue  # will get picked up in another target portfolio
```
[the remark this answers is not in the document]
Comment (tommyjcarpenter) [reply merqube_client_lib/templates/equity_baskets/multieb_create.py:83]: this is filtering out all constituents with a future or past date; could be a _lot_ of logs

File: pyproject.toml [hunk 4]
```diff
@@ -23,9 +23,9 @@ pandas_market_calendars = "*"
 requests = "*"
 
 [tool.poetry.scripts]
-create = "merqube_client_lib.templates.equity_baskets.create_index:main"
-update_portfolio = "merqube_client_lib.templates.equity_baskets.multiple_equity_basket.update_portfolio:main"
-get_eb_template = "merqube_client_lib.templates.equity_baskets.get_eb_template:main"
+create = "merqube_client_lib.templates.equity_baskets.bin.create_index:main"
+update_portfolio = "merqube_client_lib.templates.equity_baskets.in.update_portfolio:main"
```
[the remark this answers is not in the document]
Comment (tommyjcarpenter) [reply pyproject.toml]: good catch

**Review by jperozo — APPROVED** [review]

**Pull request merged**